lib/customer.rb

```diff
@@ -0,0 +1,43 @@
+require "csv"
+
+class Customer
+  attr_reader :id
+  attr_accessor :email, :address
+
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all
+    customers = []
+    CSV.read("data/customers.csv").each do |customer|
+      id = customer[0].to_i
+      email = customer[1]
+      address = customer[2..5].join(", ")
```

Comment:
It looks like you are storing addresses as strings here.  We don't actually test on this but the requirements call for a hash.  

Remember: Even our tests aren't perfect and won't catch everything, so it's a good idea to read the requirements very carefully.